### ISSUE TITLE
'scanf' causes cgdb to hang

### ISSUE THREAD

**Issue by BrandonIrizarry** [cgdb/cgdb#218]
Consider the following program:
```
/* minimal.c */
#include <stdio.h>

int main (int argc, char *argv[])
{
  int choice = 0;
  scanf("%d", &choice);
  printf("%d\n", choice);

  return 0;
}
```
Now, I run 

> $ gcc -g minimal.c
> $ gdb a.out
> (gdb) run

I enter a number, and then gdb reports something about an inferior process exiting normally. I would expect cgdb to do the same, but when I run inside it and enter an input, it hangs, and I have to abort with Control-C.

**Comment by iandinwoodie** [comment]
Thank you for the minimal, reproducible example; it's a great demonstration of the behavior. The current manual for cgdb has a ["sending I/O to the program being debugged"](https://cgdb.github.io/docs/cgdb.html#Sending-I_002fO-to-Inferior) section. For your convenience, I've included the contents of the section below:

> If the program being debugged takes input on the terminal it is recommended that the user start the program on one terminal, and attach to it with CGDB from another terminal. This is the easiest way to pass input to the debugged program.

A workflow adhering to these instructions might look like:

```bash
# Terminal 1
gcc -g minimal.c
./a.out

# Terminal 2
cgdb -- --pid=$(pidof a.out)
```

If you try this workflow, you may find that it is quite "clunky" for such a simple task. However, if we look at the [comparable section from earlier releases](http://cgdb.sourceforge.net/docs/cgdb-no-split.html#Sending-I_002fO-to-Inferior), we see that the content is much different. I've included the most relevant part below:

> This technique is similar to getting in and out of GDB mode. The tty window is not visible by default. This is because it is only needed if the user wishes to send data to the program being debugged. To display the tty window, hit T while in command mode. After hitting T you will notice that there is another window in the middle of the source window and the gdb window. This is called the tty window.

It's the behavior of this removed "tty window" feature that you seem to be expecting. You are not alone in wishing this feature would still be available (see issues #209 and #102). This feature was removed in commit 2e63f9e, therefore the last release to include the "tty window" was release v0.6.8.

Per the response in issue #209, the feature is slated for return:

> Yes. Well I should say, the next two features for cgdb are,
> - Determining a better way to allow the user to talk to the debugged program
> - Switching to GDB/MI communication
>
> I'll be thinking about the communication issue and will post some ideas to
> the mailing
> list before work is started.

**Comment by iandinwoodie** [comment]
I'm closing this issue as a duplicate of #209.
